api/plugin_helpers.go

```diff
@@ -118,6 +118,9 @@ func VaultPluginTLSProvider(apiTLSConfig *TLSConfig) func() (*tls.Config, error)
 			return nil, errwrap.Wrapf("error during api client creation: {{err}}", err)
 		}
 
+		// Reset token value to make sure nothing has been set by default
+		client.SetToken("")
```

Comment:
There's a `ClearToken()` method, should we use that?

- https://github.com/hashicorp/vault/blob/ca83be21b401b0ff35f41ae5b75f4650e5319770/api/client.go#L598

Reply:
Ah yes. Good catch!